// eslint-plugin-jest/lib/rules/prefer-to-be-null.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = void 0;

var _util = require("./util");

var _default = {
  meta: {
    docs: {
      url: (0, _util.getDocsUrl)(__filename)
    },
    messages: {
      useToBeNull: 'Use toBeNull() instead'
    },
    fixable: 'code',
    schema: []
  },

  create(context) {
    return {
      CallExpression(node) {
        const is = (0, _util.expectToBeCase)(node, null) || (0, _util.expectToEqualCase)(node, null);
        const isNot = (0, _util.expectNotToEqualCase)(node, null) || (0, _util.expectNotToBeCase)(node, null);

        if (is || isNot) {
          context.report({
            fix(fixer) {
              if (is) {
                return [fixer.replaceText((0, _util.method)(node), 'toBeNull'), fixer.remove((0, _util.argument)(node))];
              }

              return [fixer.replaceText((0, _util.method2)(node), 'toBeNull'), fixer.remove((0, _util.argument2)(node))];
            },

            messageId: 'useToBeNull',
            node: is ? (0, _util.method)(node) : (0, _util.method2)(node)
          });
        }
      }

    };
  }

};
exports.default = _default;
